### ISSUE TITLE
Unexpected token error defining constant array in php56

### ISSUE THREAD

**Issue by dario1985** [phpmd/phpmd#280]
When I try to define a constant array using the const keyword (http://php.net/manual/en/migration56.new-features.php) phpMD (version 2.2.2) generate a "Unexpected token: =>" error.

**Comment by ravage84** [comment]
Can you prepare a reduced code example that triggers that errors so we can test it?

**Comment by dario1985** [comment]
Yes, sure :)

```php
class Test
{
    const TYPE = 'type';

    const ARRAY_CONST = [
        self::TYPE => 0 // This line generate the error
    ];
}

**Comment by ravage84** [comment]
Does it work with the old array syntax?

```` php
const ARRAY_CONST = array(
    self::TYPE => 0
);
````

**Comment by manuelpichler** [comment]
This issue is related to PDepend issue pdepend/pdepend#192

**Comment by dario1985** [comment]
It trigger the same error with the old array syntax too.

```php
class Test
{
    const TYPE = 'type';

    const ARRAY_CONST = array(
        self::TYPE => 0 // This line generate the error
    );
}

**Comment by manuelpichler** [comment]
Fixed since phpmd@2.3.3
